src/main/java/ch/kalunight/zoe/repositories/LeaderboardRepository.java

```diff
@@ -0,0 +1,181 @@
+package ch.kalunight.zoe.repositories;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import ch.kalunight.zoe.model.dto.DTO;
+
+public class LeaderboardRepository {
+
+  private static final String INSERT_LEADERBOARD = "INSERT INTO leaderboard " +
+      "(lead_fk_server, lead_message_channelId, " +
+      "lead_message_id, lead_type, lead_lastrefresh) " +
+      "VALUES (%d, %d, %d, '%s', '%s') " +
+      "RETURNING lead_id";
+  
+  private static final String DELETE_LEADERBOARD_WITH_ID = "DELETE FROM leaderboard WHERE lead_id = %d";
+  
+  private static final String UPDATE_LEADERBOARD_DATA_WITH_ID = "UPDATE leaderboard SET lead_data = %s WHERE lead_id = %d";
+  
+  private static final String UPDATE_LEADERBOARD_LAST_REFRESH_WITH_ID = "UPDATE leaderboard SET lead_lastrefresh = '%s' WHERE lead_id = %d";
+  
+  private static final String SELECT_LEADERBOARD_WITH_GUILD_ID = "SELECT " + 
+      "leaderboard.lead_id, " + 
+      "leaderboard.lead_fk_server, " + 
+      "leaderboard.lead_message_channelid, " + 
+      "leaderboard.lead_message_id, " + 
+      "leaderboard.lead_type," +
+      "leaderboard.lead_data," +
+      "leaderboard.lead_lastrefresh " + 
+      "FROM server " + 
+      "INNER JOIN leaderboard ON server.serv_id = leaderboard.lead_fk_server " + 
+      "WHERE server.serv_guildid = %d";
+  
+  private static final String SELECT_LEADERBOARD_WITH_ID = "SELECT " + 
+      "leaderboard.lead_id, " + 
+      "leaderboard.lead_fk_server, " + 
+      "leaderboard.lead_message_channelid, " + 
+      "leaderboard.lead_message_id, " + 
+      "leaderboard.lead_type, " + 
+      "leaderboard.lead_data, " + 
+      "leaderboard.lead_lastrefresh " + 
+      "FROM leaderboard " + 
+      "WHERE leaderboard.lead_id = %d";
+  
+  private static final String SELECT_LEADERBOARD_WITH_LAST_REFRESH = "SELECT " + 
+      "leaderboard.lead_id, " + 
+      "leaderboard.lead_fk_server, " + 
+      "leaderboard.lead_message_channelid, " + 
+      "leaderboard.lead_message_id, " + 
+      "leaderboard.lead_type, " + 
+      "leaderboard.lead_data, " + 
+      "leaderboard.lead_lastrefresh " + 
+      "FROM leaderboard " + 
+      "WHERE leaderboard.lead_lastrefresh < '%s'";
+  
+  private LeaderboardRepository() {
+    //hide default public constructor
+  }
+  
+  public static List<DTO.Leaderboard> getLeaderboardWhoNeedToBeRefreshed() throws SQLException {
+    ResultSet result = null;
```

Comment:
Codacy found an issue: [Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.](https://app.codacy.com/manual/KaluNight/Zoe-Discord-Bot/pullRequest?prid=5821508)